sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -386,7 +386,9 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
           case relation: CatalogRelation if DDLUtils.isHiveTable(relation.catalogTable) =>
             relation.catalogTable.identifier
         }
-        EliminateSubqueryAliases(catalog.lookupRelation(tableIdentWithDB)) match {
+
+        val tableRelation = df.sparkSession.table(tableIdentWithDB).queryExecution.analyzed
```

Comment:
This change is to avoid overriding `lookupRelation` in `HiveMetastoreCatalog `, right?